### ISSUE TITLE
Date-valued field queries allow non-date values

### ISSUE THREAD

**Issue by discopatrick** [beetbox/beets#2513]
### Problem

When querying on date-valued fields, such as added and mtime, non-date values (e.g. 'abcd') are allowed, and seem to output the full collection list. For example:

```sh
$ beet -vv list mtime:abcd
no user configuration found at /Users/patrick/.config/beets/config.yaml
data directory: /Users/patrick/.config/beets
plugin paths: 
Sending event: pluginload
library database: /Users/patrick/.config/beets/library.db
library directory: /Users/patrick/Music
Sending event: library_opened
... every item in my library ...
Sending event: cli_exit
```

I'm currently working on #2506 to extend the date parser, but I feel that this issue needs to be fixed first.

### Setup

* OS: OSX 10.11.6 El Capitan
* Python version: 2.7.13
* beets version: 1.4.4 (actually this commit: https://github.com/beetbox/beets/commit/45bed808d39fae2845ba4addbd71f31be2e485d4)
* Turning off plugins made problem go away (yes/no): No, no plugins active.

My configuration (output of `beet config`) is:

```yaml
{}
```

**Comment by sampsyo** [comment]
Good call—this should probably raise an `InvalidQueryArgumentTypeError`.

**Comment by discopatrick** [comment]
I'll start work on this.

**Comment by sampsyo** [comment]
Fixed in #2517.
